tests/connectors/document_index/test_document_index.py

```diff
@@ -92,6 +99,25 @@ def is_outdated_identifier(identifier: str, timestamp_threshold: datetime) -> bo
     return not timestamp > timestamp_threshold
 
 
+def random_semantic_embed() -> EmbeddingConfig:
+    return SemanticEmbed(
+        representation=random.choice(get_args(Representation)),
```

Comment:
I would counter-argue here - if this test fails as it is now, it wouldn't be a flake, it would be a bug! Using randomization here is a nice way to increase the surface area of this test. Wherever we use this function, we effectively define the representation to be an invariant of the test. Therefore, if the test were to fail, this constraint would no longer hold and we would have found a bug.

We already make use of randomization in our other Document Index client tests, so I would be hesitant to remove it here.

Reply:
I agree that we should should cover all of the possible representations. My worry is just that by letting it pick randomly, we do not always get all of the representations (therefore potentially passing because the broken representation was not tested).  By giving it to the test parameterized,  we could always make sure that all representations are tested and also get a clear failure message which one broke.

Reply:
Hmm. After thinking about this a bit more and discussing it with the team, it probably makes sense to introduce separate tests as you suggest. Although I think it would have been fine (given there are only two variants), this way we can be more explicit.

Unfortunately it's not a simple as just parametrizing, as the existing random_index fixture also runs cleanup logic. I've implemented what I think is the best option here using a contextmanager. Hopefully this aligns with what you had in mind!

Reply:
Yes, the solution looks in line with what I had in mind 👍